jsonrpc/endpoints_debug.go

```diff
@@ -34,89 +37,188 @@ type StructLogRes struct {
-				}
-				storageRes = &storage
-			}
+func (d *DebugEndpoints) buildTraceBlock(ctx context.Context, txs []*types.Transaction, cfg *traceConfig, dbTx pgx.Tx) (interface{}, rpcError) {
+	traces := []traceBlockTransactionResponse{}
+	for _, tx := range txs {
+		traceTransaction, err := d.buildTraceTransaction(ctx, tx.Hash(), cfg, dbTx)
+		if err != nil {
+			errMsg := fmt.Sprintf("failed to get trace for transaction %v", tx.Hash().String())
+			return rpcErrorResponse(defaultErrorCode, errMsg, err)
+		}
+		traceBlockTransaction := traceBlockTransactionResponse{
+			Result: traceTransaction.(traceTransactionResponse),
+		}
+		traces = append(traces, traceBlockTransaction)
+	}
 
-			errRes := ""
-			if structLog.Err != nil {
-				errRes = structLog.Err.Error()
-			}
+	return traces, nil
+}
 
-			structLogs = append(structLogs, StructLogRes{
-				Pc:            structLog.Pc,
-				Op:            structLog.Op,
-				Gas:           structLog.Gas,
-				GasCost:       structLog.GasCost,
-				Depth:         structLog.Depth,
-				Error:         errRes,
-				Stack:         stackRes,
-				Memory:        memoryRes,
-				Storage:       storageRes,
-				RefundCounter: structLog.RefundCounter,
-			})
+func (d *DebugEndpoints) buildTraceTransaction(ctx context.Context, hash common.Hash, cfg *traceConfig, dbTx pgx.Tx) (interface{}, rpcError) {
+	traceConfig := state.TraceConfig{}
+
+	if cfg != nil {
+		traceConfig.DisableStack = cfg.DisableStack
+		traceConfig.DisableStorage = cfg.DisableStorage
+		traceConfig.EnableMemory = cfg.EnableMemory
+		traceConfig.EnableReturnData = cfg.EnableReturnData
+		traceConfig.Tracer = cfg.Tracer
+	}
+
+	result, err := d.state.DebugTransaction(ctx, hash, traceConfig, dbTx)
+	if err != nil {
+		const errorMessage = "failed to get trace"
+		log.Infof("%v: %v", errorMessage, err)
+		return nil, newRPCError(defaultErrorCode, errorMessage)
+	}
+
+	if traceConfig.Tracer != nil && *traceConfig.Tracer != "" && len(result.ExecutorTraceResult) > 0 {
+		return result.ExecutorTraceResult, nil
+	}
+
+	failed := result.Failed()
+	var returnValue interface{}
+	if traceConfig.EnableReturnData {
+		returnValue = common.Bytes2Hex(result.ReturnValue)
+	}
+
+	structLogs := d.buildStructLogs(result.StructLogs, cfg)
+
+	resp := traceTransactionResponse{
+		Gas:         result.GasUsed,
+		Failed:      failed,
+		ReturnValue: returnValue,
+		StructLogs:  structLogs,
+	}
+
+	return resp, nil
+}
+
+func (d *DebugEndpoints) buildStructLogs(stateStructLogs []instrumentation.StructLog, cfg *traceConfig) []StructLogRes {
+	structLogs := make([]StructLogRes, 0, len(stateStructLogs))
+	for _, structLog := range stateStructLogs {
+		errRes := ""
+		if structLog.Err != nil {
+			errRes = structLog.Err.Error()
 		}
 
-		resp := traceTransactionResponse{
-			Gas:         result.GasUsed,
-			Failed:      failed,
-			ReturnValue: common.Bytes2Hex(result.ReturnValue),
-			StructLogs:  structLogs,
+		op := structLog.Op
+		if op == "SHA3" {
+			op = "KECCAK256"
```

Comment:
Why? Maybe adding a comment could help

Reply:
I just followed the values returned by Geth and converted the info returned by the executor to it. Maybe we could ask the executor team "why".

Reply:
our source: https://www.evm.codes/#20?fork=berlin
![image](https://user-images.githubusercontent.com/1298871/223180427-aa578fcc-d482-43a3-91d7-3bf5ebfb2939.png)